vignettes/adtte.Rmd

```diff
@@ -0,0 +1,272 @@
+  date = DTHDT,
+  set_values_to = vars(
+    EVNTDESC = "STUDY DEATH",
+    SRCDOM = "ADSL",
+    SRCVAR = "DTHDT",
+    SRCSEQ = ASEQ
+  )
+)
+```
+
+An optional step at this stage would be required to enable derivation of duration of
+response: If using `ADRS` / `ADEVENT` parameters as input for any response dates
+(instead of a variable in `ADSL`) then you would need to use `admiral::derive_vars_merged()`
+to add the response date as a temporary variable (e.g. `TEMP_RESPDT`) to be able to feed
+into `admiral::derive_param_tte()` as the start date. You would also need to use this
+to filter the input datasets so as to only derive the records for responders.
+This could also be repeated as needed for IRF and confirmed responses.
+
+Here is an example of the code needed.
+
+```{r}
+adsl <- adsl %>%
+  derive_vars_merged(
+    dataset_add = adrs,
+    filter_add = PARAMCD == "RSP" & AVALC == "Y" & ANL01FL == "Y",
+    by_vars = vars(STUDYID, USUBJID),
+    new_vars = vars(TEMP_RESPDT = ADT)
+  )
+
+adrs <- adrs %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    by_vars = vars(STUDYID, USUBJID),
+    new_vars = vars(TEMP_RESPDT)
+  )
+```
+
+The pre-defined objects can be passed directly to `admiral::derive_param_tte()` to create
+a new time-to-event parameter. Below shows example calls for Overall Survival (OS), Progression
+Free Survival (PFS), and duration of response (as above, this is only derived for responder
+patients so we have to filter source datasets). Note that the reason for including a
+randomization date censor is to catch those patients that never have a tumor assessment.
+
+```{r}
+adtte <- derive_param_tte(
+  dataset_adsl = adsl,
+  start_date = RANDDT,
+  event_conditions = list(death_event),
+  censor_conditions = list(lastalive_censor, rand_censor),
+  source_datasets = list(adsl = adsl, adrs = adrs),
+  set_values_to = vars(PARAMCD = "OS", PARAM = "Overall Survival")
+) %>%
+  derive_param_tte(
+    dataset_adsl = adsl,
+    start_date = RANDDT,
+    event_conditions = list(pd_event, death_event),
+    censor_conditions = list(lasta_censor, rand_censor),
+    source_datasets = list(adsl = adsl, adrs = adrs),
+    set_values_to = vars(PARAMCD = "PFS", PARAM = "Progression Free Survival")
+  ) %>%
+  derive_param_tte(
+    dataset_adsl = adsl,
+    start_date = TEMP_RESPDT,
+    event_conditions = list(pd_event, death_event),
+    censor_conditions = list(lasta_censor),
+    source_datasets = list(adsl = filter(adsl, !is.na(TEMP_RESPDT)), adrs = filter(adrs, !is.na(TEMP_RESPDT))),
+    set_values_to = vars(PARAMCD = "RSD", PARAM = "Duration of Response")
+  )
+```
+```{r, echo=FALSE}
+dataset_vignette(
+  adtte,
+  display_vars = vars(USUBJID, PARAMCD, PARAM, STARTDT, ADT, CNSR)
+)
+```
+
+### Creating Your Own Time-to-Event Source Objects
+
+We advise you consult the `{admiral}` ["Creating a BDS Time-to-Event ADaM" vignette]
+(https://pharmaverse.github.io/admiral/articles/bds_tte.html) for further guidance on the
```

Comment:
The link to the vignette does not render correctly. Maybe there must be no line break between link name and URL.

Reply:
@bundfussr i tried the fix you suggested - can you let me know if works? i'm having trouble with my R session so not able to build the site from this branch to check

Reply:
It is working now.